File: CSharp/[Udemy] CSharp POO/session8/Session100Ex3/Entities/UsedProduct.cs
using System.Globalization;

namespace Session100Ex3.Entities
{
    internal class UsedProduct : Product
    {
        public DateTime ManufactureDate { get; set; }

        public UsedProduct() { }
        public UsedProduct(string name, double price, DateTime manufactureDate) : base (name, price)
        {
            ManufactureDate = manufactureDate;
        }
        public override string PriceTag()
        {
            return $"{Name} (used) $ {Price.ToString("F2", CultureInfo.InvariantCulture)} (Manufacture date: {ManufactureDate.ToShortDateString()})";
        }
    }
}

File: Java/[Udemy] Java POO/section13/Polimorfismo/src/entities/UsedProduct.java
package entities;

import java.time.LocalDate;
import java.time.format.DateTimeFormatter;

public class UsedProduct extends Product {

	private LocalDate date;

	public UsedProduct(String name, Double price, LocalDate date) {
		super(name, price);
		this.date = date;
	}

	public LocalDate getDate() {
		return date;
	}

	public void setDate(LocalDate date) {
		this.date = date;
	}
	
	@Override
    public String priceTag() {
        return String.format("%s (used) $ %.2f (Manufacture date: %s)", super.getName(), super.getPrice(), DateTimeFormatter.ofPattern("dd/MM/yyyy").format(getDate()));
    }
}
